server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -594,6 +597,30 @@ private void verifyUploadedSegments(RemoteSegmentStoreDirectory remoteSegmentSto
                 }
             }
         }
+        assertTrue(indexShard.isRemoteSegmentStoreInSync(false));
+    }
+
+    public void testRemoteSegmentStoreNotInSync() throws IOException {
```

Comment:
Do we have a test where we are able to simulate timeouts while waiting?

Reply:
Good point . Let me add that .